lib/Request_Ride_Time.dart

```diff
@@ -27,29 +66,76 @@ class _RequestRideTimeState extends State<RequestRideTime> {
 
   final List<bool> isSelected = [true, false];
 
-  Widget _buildDateField(){
-    return TextFormField(
-      decoration: InputDecoration(labelText: 'Date'),
+  String format(String date) {
+    var dates = date.split('-');
+    String formatDate = dates[1] + "/" + dates[2] + "/" + dates[0];
+    return formatDate;
+  }
+
+  _selectDate(BuildContext context) async {
+    final DateTime picked = await showDatePicker(
+      context: context,
+      initialDate: selectedDate,
+      firstDate: DateTime(2020),
+      lastDate: DateTime(2025),
+      builder: (context, child) {
+        return Theme(
+          data: ThemeData.light(),
+          child: child,
+        );
+      },
     );
+    if (picked != null && picked != selectedDate)
+      setState(() {
+        selectedDate = picked;
+        dateCtrl.text = format("$selectedDate".split(' ')[0]);
+      });
   }
 
-  Widget _buildPickupTimeField(){
+  Widget _buildPickupTimeField() {
     return TextFormField(
-      decoration: InputDecoration(labelText: 'Pickup Time'),
+      controller: pickUpCtrl,
+      decoration: InputDecoration(
+          labelText: 'Pickup Time',
+          hintText: 'Pickup Time',
+          labelStyle: TextStyle(color: Colors.black)),
+      validator: (input) {
```

Comment:
The validators don't actually get called unless you check `if (_formKey.currentState.validate())` when the button is pressed (check out step 3 of the [cookbook](https://flutter.dev/docs/cookbook/forms/validation#3-create-a-button-to-validate-and-submit-the-form)). This applies for all of the validators in the flow, but I think this can be an issue to fix in another PR where we add more validation for dates+times because right now you can actually still edit the text field and potentially enter invalid input anyway--maybe add a note about this on monday?